actionpack/test/controller/test_case_test.rb

```diff
@@ -235,10 +235,10 @@ def test_url_options_reset
   end
 
   def test_raw_post_handling
-    params = Hash[:page, { name: "page name" }, "some key", 123]
-    post :render_raw_post, params: params.dup
+    body = "Hello World"
+    post :render_raw_post, body: body
 
-    assert_equal params.to_query, @response.body
+    assert_equal body, @response.body
```

Comment:
This test pass for me on main with Rack 3.0 
Unless I am missing something, I don't think we need to change it anymore.